marimo/_islands/island_generator.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2024 Marimo. All rights reserved.
+from __future__ import annotations
+
+from textwrap import dedent
+from typing import cast
+
+from marimo import __version__, _loggers
+from marimo._ast.app import App, InternalApp
+from marimo._ast.cell import Cell, CellConfig
+from marimo._ast.compiler import compile_cell
+from marimo._output.utils import uri_encode_component
+from marimo._server.export.utils import run_app_until_completion
+from marimo._server.file_manager import AppFileManager
+
+LOGGER = _loggers.marimo_logger()
+
+
+class MarimoIslandGenerator:
+    """
+    Generates Marimo islands for embedding in other pages.
+
+    This is a great way to use another SSG framework that converts
+    Python code to HTML using marimo-islands.
+
+    Generally you will want to:
+
+    1. Find all the code snippets and add them to the generator.
+    2. Build the app.
+    3. Replace all code snippets with the rendered HTML.
+    4. Include the header in the <head> tag.
+
+    # Example
+
+    ```python
+    from marimo import MarimoIslandGenerator
+
+    generator = MarimoIslandGenerator()
+    generator.add_code("import marimo as mo")
+    generator.add_code("mo.md('Hello, islands!')")
+
+    # Build the app
+    app = await generator.build()
+
+    # Render the app
+    output = f\"\"\"
+    <html>
+        <head>
+            {generator.render_header()}
+        </head>
+        <body>
+            {generator.render("import marimo as mo")}
+            {generator.render("mo.md('Hello, islands!')")}
+        </body>
+    </html>
+    \"\"\"
+    ```
+    """
+
+    def __init__(self, app_id: str = "main"):
+        self.has_run = False
+        self._app_id = app_id
+        self._app = InternalApp(App())
+
+    def add_code(
+        self,
+        code: str,
+        *,
+        disabled: bool = False,
+        hide_code: bool = False,
+    ) -> MarimoIslandGenerator:
+        """Add a code cell to the app.
+
+        *Args:*
+
+        - code (str): The code to add to the app.
+        - disabled (bool, optional): Whether the cell should be disabled.
+            Defaults to False.
+        - hide_code (bool, optional): Whether the code/output should be hidden.
+            Defaults to False.
+        """
+
+        cell_id = self._app.cell_manager.create_cell_id()
+        cell_impl = compile_cell(code, cell_id)
+        cell_impl.configure(CellConfig(disabled=disabled, hide_code=hide_code))
```

Comment:
yea these came from the `CellConfig` which normally live in the `@app.cell(...)` args. debated consistency or diverge. im not sure if `disabled` makes sense. and then it is also kinda nice to do this later in the `render()` instead of now. 

i might remove this - not sure its worth including in the first iteration, and will add `code_block=True` and `output=True` when rendering